SU2_CFD/src/iteration_structure.cpp

```diff
@@ -3185,3 +3249,313 @@ void CDiscAdjFEAIteration::Postprocess(COutput *output,
+
+  unsigned long ExtIter = config_container[val_iZone]->GetExtIter();
+  unsigned long IntIter = 0;
+  bool unsteady = config_container[val_iZone]->GetUnsteady_Simulation() != STEADY;
+
+  if (!unsteady)
+    IntIter = ExtIter;
+  else {
+    IntIter = config_container[val_iZone]->GetIntIter();
+  }
+
+  solver_container[val_iZone][val_iInst][MESH_0][ADJHEAT_SOL]->ExtractAdjoint_Solution(geometry_container[val_iZone][val_iInst][MESH_0],
+                                                                                       config_container[val_iZone]);
+}
+
+void CDiscAdjHeatIteration::InitializeAdjoint(CSolver *****solver_container,
+                                              CGeometry ****geometry_container,
+                                              CConfig **config_container,
+                                              unsigned short iZone, unsigned short iInst){
+
+  /*--- Initialize the adjoints the conservative variables ---*/
+
+  solver_container[iZone][iInst][MESH_0][ADJHEAT_SOL]->SetAdjoint_Output(geometry_container[iZone][iInst][MESH_0],
+                                                                         config_container[iZone]);
+}
+
+
+void CDiscAdjHeatIteration::RegisterInput(CSolver *****solver_container,
+                                          CGeometry ****geometry_container,
+                                          CConfig **config_container,
+                                          unsigned short iZone, unsigned short iInst,
+                                          unsigned short kind_recording){
+
+  if (kind_recording == FLOW_CONS_VARS || kind_recording == COMBINED){
+
+    /*--- Register flow and turbulent variables as input ---*/
+
+    solver_container[iZone][iInst][MESH_0][ADJHEAT_SOL]->RegisterSolution(geometry_container[iZone][iInst][MESH_0], config_container[iZone]);
+
+    solver_container[iZone][iInst][MESH_0][ADJHEAT_SOL]->RegisterVariables(geometry_container[iZone][iInst][MESH_0], config_container[iZone]);
+
+  }
+  if (kind_recording == MESH_COORDS){
+
+    /*--- Register node coordinates as input ---*/
+
+    geometry_container[iZone][iInst][MESH_0]->RegisterCoordinates(config_container[iZone]);
+
+  }
+}
+
+void CDiscAdjHeatIteration::SetDependencies(CSolver *****solver_container,
+                                            CGeometry ****geometry_container,
+                                            CConfig **config_container,
+                                            unsigned short iZone, unsigned short iInst,
+                                            unsigned short kind_recording){
+
+  if ((kind_recording == MESH_COORDS) || (kind_recording == NONE)  ||
+      (kind_recording == GEOMETRY_CROSS_TERM) || (kind_recording == ALL_VARIABLES)){
+
+    /*--- Update geometry to get the influence on other geometry variables (normals, volume etc) ---*/
+
+    geometry_container[iZone][iInst][MESH_0]->UpdateGeometry(geometry_container[iZone][iInst], config_container[iZone]);
+
+  }
+
```

Comment:
Hm. I still don't understand.
As the routines (or their contents) have to be called there to be recorded, I don't know how one could move them to the solver level. Or do you think of some kind of wrapping?

Reply:
Yes wrapping, i.e. directSolver->SetDependencies() so that if in the future the list of dependencies increases only one place in the code needs updating.
Or a comment on both areas making reference to the other.

Reply:
Ok. 
Me personally, I prefer four explicit calls over an additional routine wrapping just these four, also in terms of code readability.
(Having a SetDependencies-routine on solver level would also imply that one fixes or pre-defines the contents independently of the physical domain, whereas the iterator might be the more appropriate place for that.) What do you think?

Reply:
It does not force that because you still have the iterations to define specifics.
Also having such a method at solver level gives you a place to call methods that only that solver declares, and that in general avoids the explosion of base class methods.
But my main point was to avoid duplication (I'm a pedant I know) or to at least to alert people about it.

Reply:
Oh no worries, it's nice that we start having discussions here. It also serves the transparency :+1: 

> It does not force that because you still have the iterations to define specifics.
True, but I have doubts whether that simplifies things after all.

> But my main point was to avoid duplication (I'm a pedant I know) or to at least to alert people about it.
Well both iterators need them, so it's just about calling the routines directly or through a wrapper.

Reply:
I agree that we should move all these things to specific routines in the solver. However, we should also do it for the turbulence (and transition) solvers. So I would move that to a later PR ...

Reply:
OK. Democratic decision adopted :+1: 
I will take care of wrapping them for the heat solver and then have a look to which PR they might fit.